From 89af26242515a62796cf609e9eb26ba14689a942 Mon Sep 17 00:00:00 2001
From: Haikel Guemar <hguemar@fedoraproject.org>
Date: Fri, 25 Mar 2016 10:42:29 +0100
Subject: [PATCH] Support new mariadb-server-galera package

mariadb-server-galera is now a subpackage of mariadb-server
and requires it so we can't remove it
https://bugzilla.redhat.com/show_bug.cgi?id=1310622

Change-Id: I25de6e65228257fd39df0d9ad59a56e331f16393
---
 packstack/puppet/modules/packstack/lib/facter/mariadb.rb | 14 ++++++++++++++
 packstack/puppet/templates/mariadb_install.pp            | 16 +++++++++++++---
 2 files changed, 27 insertions(+), 3 deletions(-)
 create mode 100644 packstack/puppet/modules/packstack/lib/facter/mariadb.rb

diff --git a/packstack/puppet/modules/packstack/lib/facter/mariadb.rb b/packstack/puppet/modules/packstack/lib/facter/mariadb.rb
new file mode 100644
index 0000000..f2c35f2
--- /dev/null
+++ b/packstack/puppet/modules/packstack/lib/facter/mariadb.rb
@@ -0,0 +1,14 @@
+
+# Check if mariadb provides galera server
+
+Facter.add(:mariadb_provides_galera) do
+  setcode do
+    if Facter.value(:operatingsystem) == 'Fedora' and Facter.value(:operatingsystemmajrelease).to_i > 22
+      command = 'dnf repoquery --whatprovides mariadb-galera-server'
+    else
+      command = 'repoquery --whatprovides mariadb-galera-server'
+    end
+    output = Facter::Util::Resolution.exec(command)
+    (output =~ /mariadb-server-galera.*/) != nil
+  end
+end
diff --git a/packstack/puppet/templates/mariadb_install.pp b/packstack/puppet/templates/mariadb_install.pp
index 2a001e3..8d5fd20 100644
--- a/packstack/puppet/templates/mariadb_install.pp
+++ b/packstack/puppet/templates/mariadb_install.pp
@@ -1,9 +1,19 @@
 
 $max_connections = $service_workers * 128
 
-# Package mariadb-server conflicts with mariadb-galera-server
+
+if ($::mariadb_provides_galera == 'true') {
+  # Since mariadb 10.1 galera is included in main mariadb
+  $mariadb_package_name = 'mariadb-server-galera'
+  $mariadb_present      = 'present'
+} else  {
+  # Package mariadb-server conflicts with mariadb-galera-server
+  $mariadb_package_name = 'mariadb-galera-server'
+  $mariadb_present      = 'absent'
+}
+
 package { 'mariadb-server':
-  ensure => absent,
+  ensure => $mariadb_present,
 }
 
 $bind_address = hiera('CONFIG_IP_VERSION') ? {
@@ -15,7 +25,7 @@ $bind_address = hiera('CONFIG_IP_VERSION') ? {
 $mysql_root_password = hiera('CONFIG_MARIADB_PW')
 
 class { '::mysql::server':
-  package_name     => 'mariadb-galera-server',
+  package_name     => $mariadb_package_name,
   restart          => true,
   root_password    => $mysql_root_password,
   require          => Package['mariadb-server'],
-- 
2.4.3

